lib/Dialects/VectorExt/Transform/VectorMaskingUtils.cpp

```diff
@@ -0,0 +1,138 @@
+//===- VectorMaskingUtils.cpp - Utilities for vector masking --------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Dialects/VectorExt/VectorMaskingUtils.h"
+
+#include "Dialects/VectorExt/VectorExtOps.h"
+#include "mlir/Dialect/Linalg/IR/LinalgOps.h"
+#include "mlir/Dialect/Vector/VectorOps.h"
+#include "mlir/IR/AffineMap.h"
+#include "mlir/Support/LogicalResult.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallPtrSet.h"
+
+using namespace mlir;
+using namespace mlir::linalg;
+using namespace mlir::vector;
+using namespace mlir::vector_ext;
+
+using ExprList = ArrayRef<ArrayRef<AffineExpr>>;
+
+/// Move the operation range `opRange` before operation `dest`.
+static void moveOperationsBefore(llvm::iterator_range<Block::iterator> opRange,
+                                 Operation *dest) {
+  if (opRange.empty())
+    return;
+  dest->getBlock()->getOperations().splice(
+      Block::iterator(dest), opRange.begin()->getBlock()->getOperations(),
+      opRange.begin(), opRange.end());
+}
+
+/// Iterate over all the definitions produced by the operation range `opRange`
+/// and gather those with users outside the range.
+static void
+getDefsWithExternalUses(llvm::iterator_range<Block::iterator> opRange,
```

Comment:
I think we consistently use the tern "outside" in MLIR.
So `getDesfsWithUsesOutside` ?